lib/queue.rb

```diff
@@ -1,31 +1,69 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new()
+    @max = 20
+    @front = 0
+    @back = 0
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @back == @front-1 || (@front == 0 && @back == @max - 1)
+      # account for full Queue
+      return raise ArgumentError.new("Queue is full")
+    elsif self.empty?
+      @front = 0
+      @back = 0
+    elsif @back == @max-1
+      @back = 0 # if last then circles around
+    else 
+      @back = @back + 1 # incr by 1
+    end
+    @store[@back] = element
+
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if self.empty?
+      raise ArgumentError.new("Queue's Empty") 
+    end
+
+    value = @store[@front]
+    @store[@front] = nil
+
+    if @front == @max - 1 # circle
+      @front = 0
+    else
+      @front = @front + 1
+    end
+    return value
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
 
   def size
-    raise NotImplementedError, "Not yet implemented"
+    return 0 if @front == -1
+    if @front < @back
+      @back - @front + 1
+    else
+      @max - (@back - @front)
+    end
   end
 
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    # return @front == -1
+    return @store.all?{ |e| e.nil? }
   end
```

Comment:
Honestly, didn't think of it.  All I knew was I wanted to make sure the list was empty and I was getting errors because of nil  so I googled for a fix. :-(